### PR TITLE
Fixes #24048 - Add API to return Pulp URL

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ekohl** [comment]
I don't see the context for why this is needed. Isn't this an internal detail for the smart proxy?

**Comment by ehelms** [comment]
There was a follow up PR to Katello itself to help explain the change (and how to use it). TLDR is that the smart proxy should be the source of truth for declaring where Pulp is located (e.g. when Pulp and Smart Proxy are not co-located on the same host or with containers). More details:

https://github.com/Katello/katello/pull/7470

**Comment by ekohl** [comment]
In the HA proxy RFC (https://community.theforeman.org/t/highly-available-smart-proxies-part-2/8299) the aspect of the "customer"-facing side and foreman-side is addressed as well. Your current PRs appear to make the situation a bit better so it shouldn't block it, but it is good to keep in mind.

**Comment by ekohl** [comment]
I suspect the tests also fail on Ruby 2.5 without this patch.

**Comment by ehelms** [comment]
I think this change is valid no matter what the outcome of the Katello usage is since it provides a way to have a smart proxy and Pulp at different hostnames. @ekohl what to do with the failures?

File: lib/smart_proxy_pulp_plugin/pulp_api.rb [hunk 1]
```diff
@@ -35,5 +35,14 @@ class Api < Sinatra::Base
       content_type :json
       {:puppet_content_dir => ::PulpProxy::Settings.settings.puppet_content_dir}.to_json
     end
+
+    get '/status/settings' do
+      content_type :json
+      settings = {
```
Comment (ekohl) [inline lib/smart_proxy_pulp_plugin/pulp_api.rb]: Should we add the `puppet_content_dir` too and deprecate the `/status/puppet` endpoint?

Comment (ehelms) [reply lib/smart_proxy_pulp_plugin/pulp_api.rb]: I suppose that makes sense.

**Comment by ekohl** [comment]
The ruby 2.5 failure is because of the URLs: `https://localhost/pulp//api/v2/status/` (note the double /) appears to have been normalized on 2.4 and earlier.

**Comment by ehelms** [comment]
@ekohl Thanks for that fix. This is passing now, though I am not sure how to officially deprecate an API in the smart proxy.

**Comment by ekohl** [comment]
@ehelms AFAIK there's no official way but perhaps an inline comment would be good enough for now?

File: lib/smart_proxy_pulp_plugin/pulp_api.rb [hunk 2]
```diff
@@ -31,9 +31,20 @@ class Api < Sinatra::Base
       end
     end
 
+    # DEPRECATED: Use /settings
```
Comment (ehelms) [inline lib/smart_proxy_pulp_plugin/pulp_api.rb]: @ekohl added here

File: lib/smart_proxy_pulp_plugin/pulp_api.rb [hunk 3]
```diff
@@ -31,9 +31,20 @@ class Api < Sinatra::Base
...
     get '/status/puppet' do
       content_type :json
       {:puppet_content_dir => ::PulpProxy::Settings.settings.puppet_content_dir}.to_json
     end
+
+    get '/status/settings' do
```
Comment (ehelms) [inline lib/smart_proxy_pulp_plugin/pulp_api.rb]: I was debating if this should just be `/settings` ?

**Comment by jturel** [comment]
Can this be merged to unblock https://github.com/Katello/katello/pull/7470 ?

**Comment by ehelms** [comment]
@jturel This is not that simple, and kinda adds a one off solution we were hoping to avoid. I have not come back to it in a while due to other conflicts.

**Pull request closed without merging**